### PR TITLE
Mark aliasing property of error

### LINKED ISSUE #102: Julia crash with Enzyme on product of sums
Enzyme (current master) crashes Julia when auto-diffing a product of sums.

A single sum works fine:

```julia
using Enzyme

x = rand(10)
dx = zero(x)

foo(x::AbstractVector{<:Real}) = sum(x)
foo(x)
autodiff(foo, Duplicated(x, dx))
```

But

```julia
bar(x::AbstractVector{<:Real}) = sum(x) * sum(x)
bar(x)
autodiff(bar, Duplicated(x, dx))
```

goes boom:

```
julia: /buildworker/worker/package_linux64/build/src/llvm-late-gc-lowering.cpp:857: std::vector<int, std::allocator<int> > LateLowerGCFrame::NumberAllBase(State&, llvm::Value*): Assertion `Tracked.size() == Numbers.size()' failed.

signal (6): Aborted
in expression starting at REPL[9]:1
gsignal at /lib64/libc.so.6 (unknown line)
abort at /lib64/libc.so.6 (unknown line)
__assert_fail_base at /lib64/libc.so.6 (unknown line)
__assert_fail at /lib64/libc.so.6 (unknown line)
...
Aborted (core dumped)
```

### PR DESCRIPTION
**Pull request by wsmoses**
Resolves #102

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -126,6 +126,17 @@ end
     @test dinp ≈ Float64[1.0, 1.0]
 end
 
+@testset "Advanced array tests sq" begin
+    function arsumsq(f::Array{T}) where T
+        return sum(f) * sum(f)
+    end
+    inp = Float64[1.0, 2.0]
+    dinp = Float64[0.0, 0.0]
+    autodiff(arsum2, Active, Duplicated(inp, dinp))
```
Comment (vchuravy) [inline test/runtests.jl:135]:
```suggestion
    autodiff(arsumsq, Active, Duplicated(inp, dinp))
```

**Comment by vchuravy** [comment]
Rebase?

File: src/compiler.jl [hunk 2]
```diff
@@ -4044,6 +4044,15 @@ function GPUCompiler.codegen(output::Symbol, job::CompilerJob{<:EnzymeTarget};
                                   ])
             continue
         end
+        # Since this is noreturn, it can't write to any operations in the function
+        # in a way accessible by the function. Ideally the attributor should actually
```
Comment (vchuravy) [inline src/compiler.jl:4048]: This is not generally true for no-return functions in the presence of longjump/setjmp. It is true for this function

File: src/compiler.jl [hunk 3]
```diff
@@ -4044,6 +4044,15 @@ function GPUCompiler.codegen(output::Symbol, job::CompilerJob{<:EnzymeTarget};
                                   ])
             continue
         end
+        # Since this is noreturn, it can't write to any operations in the function
```
Comment (wsmoses) [inline src/compiler.jl:4047]:
```suggestion
        # Since this is noreturn and it can't write to any operations in the function
```

**Pull request merged**